internal/storage/storage.go

```diff
@@ -170,6 +204,10 @@ func (sb *s3Backend) NewFileReader(filePath string) (io.ReadCloser, error) {
 
 // NewFileWriter uploads the contents of an io.Reader to a S3 bucket
 func (sb *s3Backend) NewFileWriter(filePath string) (io.WriteCloser, error) {
+	if sb == nil {
+		return nil, fmt.Errorf("Invalid s3Backend")
+	}
+
```

Comment:
Is this necessary?An invalid back-end should have been caught at startup and prevented this point from being reached.